packages/http-server/src/__tests__/server.oas.spec.ts

```diff
@@ -11,14 +11,39 @@ function checkErrorPayloadShape(payload: string) {
   expect(parsedPayload).toHaveProperty('detail');
 }
 
+async function instantiatePrism(specPath: string) {
+  const server = createServer({}, { components: {}, config: { mock: { dynamic: false } } });
+  await server.prism.load({
+    path: relative(process.cwd(), specPath),
+  });
+  return server;
+}
+
+describe('given getOperationWithBody.oas2.json', () => {
+  test('when GET /pet with invalid body returns correct error message', async () => {
```

Comment:
so `when` should go to `describe` as it describes the conditions, but maybe there is some agreement on this I'm not aware of?

Reply:
`when` doesn't describe the _condition_ but the _action_.
`given` describes the condition
So for instance
`given` condition
`when` action
`then` result

Reply:
yeah, sure it would make sense if we were using cucumber. So I understand that's agreed to put `when` inside `test`/`it`?

Reply:
It’s not about wether we are using cucumber or not, describe it syntax works the same in any test suite. I’ve used describe-it syntax with rspec, goconvey, kahlan, mocha, jasmine, etc. 

Chris your comment looks good but what you’ve put into the test suite looks incorrect according to the guide you just set out. 

Let’s take cucumber as a guide https://cucumber.io/docs/guides/overview/

But you don’t need to write “test(then” because you can use “it (does a thing” 

That’s what the it function is there for :)

Reply:
You definitely don’t want test(when

The it’s are assertions, setup goes elsewhere.